src/units/abilities.cpp

```diff
@@ -1205,15 +1257,52 @@ std::pair<std::string, std::string> attack_type::damage_type() const
 		return {type(), ""};
 	}
 
-	std::optional<std::string> replacement_type = select_damage_type(abil_list, "replacement_type");
-	std::optional<std::string> alternative_type = select_damage_type(abil_list, "alternative_type");
+	unit_ability_list resistance_abilities;
+	if(other_){
+		resistance_abilities = (*other_).get_abilities_weapons("resistance", other_loc_, other_attack_, shared_from_this());
+	}
+	std::optional<std::string> replacement_type = select_damage_type(abil_list, "replacement_type", resistance_abilities);
+	std::optional<std::string> alternative_type = select_damage_type(abil_list, "alternative_type", resistance_abilities);
 	std::string type_damage = replacement_type.value_or(type());
 	if(alternative_type && type_damage != *alternative_type){
 		return {type_damage, *alternative_type};
 	}
 	return {type_damage, ""};
 }
 
+std::vector<std::string> attack_type::damage_alternative_type() const
+{
+	unit_ability_list abil_list = get_specials_and_abilities("damage_type");
+	if(abil_list.empty()){
+		return {};
+	}
+	std::unordered_map<std::string, unsigned int> type_count;
+	for(auto& i : abil_list) {
+		const config& c = *i.ability_cfg;
+		if(!c["alternative_type"].empty()) {
+			std::string type = c["alternative_type"].str();
+			++type_count[type];
+		}
+	}
+
+	if (type_count.empty()) return {};
+
+	std::vector<std::string> type_list;
+	for(auto& i : type_count){
+		type_list.push_back(i.first);
+	}
+
+	if(type_list.size() > 1){
+		std::sort(type_list.begin(), type_list.end());
+	}
+	if(type_list.size() <= 5){
```

Comment:
No, because where put argument else in an [damage_type]? For reuse your argument, i put argument miit to 5 for limit  showing issue or not limit but not something what what wml develloper must decide

Reply:
i wiil remove limit

Reply:
Yes, the WML developer has nothing to do with this at all.

Reply:
limit removed